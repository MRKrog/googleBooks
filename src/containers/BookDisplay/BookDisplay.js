import React, { Component } from 'react';
import Book from '../Book/Book';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

export class BookDisplay extends Component {
  render() {
    const { searchedBooks } = this.props;
    return (
      <div className="BookDisplay">
        {
          searchedBooks.map(singleBook => (
            <Book key={singleBook.id}
                  id={singleBook.id}
                  title={singleBook.title}
                  author={singleBook.author}
                  image={singleBook.image}
                  publisher={singleBook.publisher}
                  savedStatus={singleBook.savedStatus}
              />
            )
          )}
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  searchedBooks: state.searchedBooks,
})

BookDisplay.propTypes = {
  searchedBooks: PropTypes.array.isRequired,
}

export default connect(mapStateToProps, null)(BookDisplay);
