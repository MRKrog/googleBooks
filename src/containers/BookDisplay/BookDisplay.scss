.BookDisplay {
  display: flex;
  flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  justify-content: center;
  padding: 1em;
  align-items: center;
}
