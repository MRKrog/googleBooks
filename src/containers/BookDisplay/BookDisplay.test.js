import React, { Component } from 'react';
import { BookDisplay, mapStateToProps } from './BookDisplay';
import { shallow } from 'enzyme';

const mockSearchedBooks = [
  {
    id:"az8pDwAAQBAJ",
    title:"The Science of Cooking",
    author:"Stuart Farrimond",
    image:"http://books.google.com/books/content?id=az8pDwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    publisher:"Penguin",
    savedStatus:false,
  },
  {
    id:"IFiIg7elnwIC",
    title:"Cooking in Ancient Civilizations",
    author:"Cathy K. Kaufman",
    image:"http://books.google.com/books/content?id=IFiIg7elnwIC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    publisher:"Greenwood Publishing Group",
    savedStatus:false,
  }
]

describe('BookDisplay', () => {

  describe('BookDisplay Component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        <BookDisplay searchedBooks={mockSearchedBooks}/>
      )
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

  })

  describe('mapStateToProps', () => {

  });

})
