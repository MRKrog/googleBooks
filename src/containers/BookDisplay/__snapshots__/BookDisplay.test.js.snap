// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDisplay BookDisplay Component should match the snapshot 1`] = `
<div
  className="BookDisplay"
>
  <Connect(Book)
    author="Stuart Farrimond"
    id="az8pDwAAQBAJ"
    image="http://books.google.com/books/content?id=az8pDwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
    key="az8pDwAAQBAJ"
    publisher="Penguin"
    savedStatus={false}
    title="The Science of Cooking"
  />
  <Connect(Book)
    author="Cathy K. Kaufman"
    id="IFiIg7elnwIC"
    image="http://books.google.com/books/content?id=IFiIg7elnwIC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
    key="IFiIg7elnwIC"
    publisher="Greenwood Publishing Group"
    savedStatus={false}
    title="Cooking in Ancient Civilizations"
  />
</div>
`;
